lms/djangoapps/teams/views.py

```diff
@@ -0,0 +1,448 @@
+            get_course(course_key)
+        except InvalidKeyError:
+            field_errors['course_id'] = {
+                'developer_message': "course_id is not valid.",
+                'user_message': _("course_id is not valid"),
+            }
+        except ValueError:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        if course_key and not (CourseEnrollment.is_enrolled(request.user, course_key) or request.user.is_staff):
+            return Response({'detail': "User must be enrolled."}, status=status.HTTP_403_FORBIDDEN)
+
+        data = request.DATA
+        data['course_id'] = course_key
+
+        serializer = CourseTeamCreationSerializer(data=data)
+        add_serializer_errors(serializer, data, field_errors)
+
+        if field_errors:
+            return Response({
+                'field_errors': field_errors,
+            }, status=status.HTTP_400_BAD_REQUEST)
+        else:
+            team = serializer.save()
+            return Response(CourseTeamSerializer(team).data)
+
+
+class TeamsDetailView(RetrievePatchAPIView):
+    """
+        **Use Cases**
+
+            Get or update a course team's information. Updates are supported
+            only through merge patch.
+
+        **Example Requests**:
+
+            GET /api/team/v0/teams/{team_id}}
+
+            PATCH /api/team/v0/teams/{team_id} "application/merge-patch+json"
+
+        **Response Values for GET**
+
+            If the user is logged in, the response contains the following fields:
+
+            * id: The team's unique identifier.
+
+            * name: The name of the team.
+
+            * is_active: True if the team is currently active. If false, the team is considered "soft deleted" and
+              will not be included by default in results.
+
+            * course_id: The identifier for the course this team belongs to.
+
+            * topic_id: Optionally specifies which topic the team is associated with.
+
+            * date_created: Date and time when the team was created.
+
+            * description: A description of the team.
+
+            * country: Optionally specifies which country the team is associated with.
+
+            * language: Optionally specifies which language the team is associated with.
+
+            * membership: A list of the users that are members of the team. See membership endpoint for more detail.
+
+            For all text fields, clients rendering the values should take care
+            to HTML escape them to avoid script injections, as the data is
+            stored exactly as specified. The intention is that plain text is
+            supported, not HTML.
+
+            If the user is not logged in, a 403 error is returned.
+
+            If the specified team does not exist, a 404 error is returned.
+
+        **Response Values for PATCH**
+
+            Only staff can patch teams. If the specified team does not exist, a
```

Comment:
Did I get this backwards? In the User API, staff get a 403 to tell them that they are not allowed to patch, while regular users get a 404 so they can't tell whether or not the resource exists. In this situation, folks who know that it exists should get a 403 (staff and user's enrolled in the course), and everyone else should get a 404. If we don't really care about leaking the existence of teams, then everybody could get a 403.